screens/LoginScreens/SignUpScreen.jsx

```diff
@@ -1,13 +1,119 @@
-import { Text, View, Button } from "react-native";
+import {
+    Text,
+    View,
+    Button,
+    TextInput,
+    KeyboardAvoidingView,
+    ActivityIndicator,
+} from "react-native";
+import React, { useState } from "react";
+import { createUserWithEmailAndPassword, updateProfile } from "firebase/auth";
+import { firebase_auth } from "../../firebase.config.js";
+import FormButton from "../../components/FormButton.jsx";
 
 export default function SignUpScreen({ navigation }) {
+    const [firstName, setFirstName] = useState("");
+    const [lastName, setLastName] = useState("");
+    const [email, setEmail] = useState("");
+    const [password, setPassword] = useState("");
+    const [confirmPassword, setConfirmPassword] = useState("");
+    const [passwordMissmatch, setPasswordMismatch] = useState(false);
+    const [loading, setLoading] = useState(false);
+    const auth = firebase_auth;
+
+    const handleSignUp = async () => {
+        setLoading(true);
+
+        if (passwordMissmatch) {
+            console.log("Passwords do not match");
+            alert("Passwords do not match");
+            return;
+        }
+
+        try {
+            const response = await createUserWithEmailAndPassword(auth, email, password);
+            await updateProfile( response.user, {displayName: `${firstName} ${lastName}`} );
+            console.log(response);
+        } catch (error) {
+            console.log(error);
+            alert(`Sign up failed: ${error.message}`);
+        } finally {
+            setLoading(false);
+        }
+    };
+
     return (
-        <View>
-            <Text>
-                Sign Up
-            </Text>
-            <Button title = "Back to Login"
-                onPress = {() => navigation.navigate("Login")} />
+        <View className="flex-1 mx-10 justify-center">
+            <KeyboardAvoidingView behavior="position">
+                <TextInput
+                    className="px-5 py-2 my-1 bg-slate-100 rounded-xl border-2"
+                    value={firstName}
+                    placeholder="First Name"
+                    autoCapitalize="words"
+                    onChangeText={(text) => setFirstName(text.replace(/\s/g, ""))}
+                    blurOnSubmit={false}
+                />
+                <TextInput
+                    className="px-5 py-2 my-1 bg-slate-100 rounded-xl border-2"
+                    value={lastName}
+                    placeholder="Last Name"
+                    autoCapitalize="words"
```

Comment:
What is the default for this btw?

Reply:
sentences is default

Reply:
I mean depends which but yh

Reply:
think cos there cant be spaces, sentences will be same outcome as words so yh ill just go default